src/cargo/core/manifest.rs

```diff
@@ -384,7 +384,10 @@ impl Target {
         self
     }
     pub fn set_harness(&mut self, harness: bool) -> &mut Target {
-        self.harness = harness;
+        match  self.kind {
+            TargetKind::Lib(_) => (),
+            _ => self.harness = harness,
+        }
```

Comment:
Hm I may be forgetting now, but this should be ok, right? If we pass `--cfg test` then a library can have its own custom test harness/main function, so it seems legitimate that `harness = false` could work for a library?